NationStatesAPIBot/NationStatesAPIBot/Manager/PermissionManager.cs

```diff
@@ -38,7 +38,14 @@ public Task RemoveRoleAsync(Role role, BotDbContext dbContext)
 
         public Task RevokePermissionAsync(string discordUserId, Permission permission, BotDbContext dbContext)
         {
-            throw new NotImplementedException();
+            return Task.Run(() =>
+            {
+                var user = dbContext.Users.Single(u => u.DiscordUserId == discordUserId);
+                var perm = user.UserPermissions.Find(p => p.Permission.Id == permission.Id);
+                var update = dbContext.Users.Update(user);
```

Comment:
Why are you updating the user here?

Reply:
The description of the Update method said: "Begins tracking the given entity in the Modified state such that it will be updated in the database when SaveChanges() is called.", so I assume that means "Update" doesn't update it, but instead merely marks it as being modified. The next line would then do the actual updating of information, before the changes are saved with SaveChanges. I may have interpreted it wrong though.

Reply:
It just confused me a bit "Update" before actually changing it. Alright. :smile: